src/LanguageServer/Protocol/Features/Diagnostics/EngineV2/DiagnosticIncrementalAnalyzer_GetDiagnostics.cs

```diff
@@ -108,14 +111,16 @@ async Task<ImmutableDictionary<DiagnosticAnalyzer, DiagnosticAnalysisResult>> Ge
                 // If there was a 'ForceAnalyzeProjectAsync' run for this project, we can piggy back off of the
                 // prior computed/cached results as they will be a superset of the results we want.
                 //
-                // Note: the caller will loop over *its* analzyers, grabbing from the full set of data we've cached
-                // for this project, and filtering down further.  So it's ok to return this potentially larger set.
+                // Note: the caller will loop over *its* analyzers, grabbing from the full set of data we've cached for
+                // this project, and filtering down further.  So it's ok to return this potentially larger set.
                 //
                 // Note: While ForceAnalyzeProjectAsync should always run with a larger set of analyzers than us
                 // (since it runs all analyzers), we still run a paranoia check that the analyzers we care about are
                 // a subset of that call so that we don't accidentally reuse results that would not correspond to
                 // what we are computing ourselves.
-                if (_projectToForceAnalysisData.TryGetValue(project, out var box) &&
+                var checksum = await project.GetDependentChecksumAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
>var checksum = await project.GetDependentChecksumAsync(cancellationToken).ConfigureAwait(false); [](http://example.com/codeflow?start=16&length=96)

Not sure of the expense of this, but if would it make sense to have this only done if the TryGetValue succeeds?

Reply:
so, if TryGet fails, we need this data to compute and cache.   If it succeed, we need it to compare with the cached value.  So we always need it.